core/types/types.cc

```diff
@@ -367,15 +382,17 @@ void LiteralType::_sanityCheck(const GlobalState &gs) const {
 }
 
 bool LiteralType::equals(const LiteralType &rhs) const {
-    if (this->value != rhs.value) {
```

Comment:
oh no was this doing bitwise floating point comparison

Reply:
SURE WAS